### PR TITLE
Add interactions to Adventure

### PR DESCRIPTION
**Pull request by TrustyJAID**
This PR is intending to add full interactions support to Adventure.

Planned changes:
 - [x] Add buttons to main adventure game
 - [x] Convert existing menus to buttons and select menus
 - [x] Introduce slash commands
 - [x] Make the cart use buttons instead of reactions

### CODE REVIEW

File: adventure/helpers.py [hunk 1]
```diff
@@ -25,26 +25,54 @@ def escape(t: str) -> str:
...
+                    colour = await cog.bot.get_embed_colour(ctx.channel)
+                else:
+                    colour = await ctx.embed_colour()
             embed = discord.Embed(description=message, color=colour)
             if image:
                 embed.set_thumbnail(url=image)
-            return await ctx.send(embed=embed)
+            if is_slash:
+                if ctx.response.is_done():
+                    await ctx.followup.send(embed=embed, ephemeral=ephemeral)
```
Comment (Drapersniper) [inline adventure/helpers.py:54]: should we be using the wait kwag on followup.send ? the reason I ask is that I noticed this is the default on dpy.py for an interaction converted to context when you call ctx.send().

i also noticed not using the wait kwarg arg the response doesn't get shown to the user straight away when I had it off for one of my interaction the message didn't get sent until a while later.

Comment (TrustyJAID) [reply adventure/helpers.py:54]: The "wait" kwarg on followup.send just returns the message from the send. in this case I don't think I actually care.

**Comment by ltzmax** [comment]
Hey. while ive been running this version of adventure for quite along time i was thinking this might help to post an error ive got in logs.
```py
Apr 21 21:25:18 voltrabot python[2058345]: [2023-04-21 21:25:18] [ERROR] discord.ui.view: Ignoring exception in view <GameSession timeout=120 children=6> for item <SpecialActionButton style=<ButtonStyle.primary: 1> url=None disabled=False label='Special Action' emoji=<PartialEmoji animated=False name='⚛️' id=None> row=None>
Apr 21 21:25:18 voltrabot python[2058345]: Traceback (most recent call last):
Apr 21 21:25:18 voltrabot python[2058345]:   File "/home/max/redenv/lib/python3.11/site-packages/discord/ui/view.py", line 427, in _scheduled_task
Apr 21 21:25:18 voltrabot python[2058345]:     await item.callback(interaction)
Apr 21 21:25:18 voltrabot python[2058345]:   File "/home/max/redbot/cogs/CogManager/cogs/adventure/game_session.py", line 754, in callback
Apr 21 21:25:18 voltrabot python[2058345]:     await self.send_focus(interaction, c)
Apr 21 21:25:18 voltrabot python[2058345]:   File "/home/max/redbot/cogs/CogManager/cogs/adventure/game_session.py", line 686, in send_focus
Apr 21 21:25:18 voltrabot python[2058345]:     await smart_embed(
Apr 21 21:25:18 voltrabot python[2058345]:   File "/home/max/redbot/cogs/CogManager/cogs/adventure/helpers.py", line 73, in smart_embed
Apr 21 21:25:18 voltrabot python[2058345]:     msg = await interaction.original_message()
Apr 21 21:25:18 voltrabot python[2058345]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 21 21:25:18 voltrabot python[2058345]: AttributeError: 'Interaction' object has no attribute 'original_message'
```

Any other thing is `[p]heroclass [name-of-class]` also does not allow you to choose a class it seems.

**Comment by ltzmax** [comment]
This as well seems to happen when doing `[p]heroclass ranger`
[from sentry](https://atticusbot.sentry.io/share/issue/0ab97984b3244a379312a5e796ecb88d/)
```py
CommandOnCooldown: You are on cooldown. Try again in 7146.69s
  File "discord/ext/commands/bot.py", line 1350, in invoke
    await ctx.command.invoke(ctx)
  File "discord/ext/commands/core.py", line 1015, in invoke
    await self.prepare(ctx)
  File "redbot/core/commands/commands.py", line 480, in prepare
    self._prepare_cooldowns(ctx)
  File "discord/ext/commands/core.py", line 914, in _prepare_cooldowns
    raise CommandOnCooldown(bucket, retry_after, self._buckets.type)  # type: ignore

  File "discord/ext/commands/core.py", line 229, in wrapped
    ret = await coro(*args, **kwargs)
  File "adventure/class_abilities.py", line 107, in heroclass
    currency_name=currency_name, clz=clz.ansi(classes[clz.class_name]["name"])
                                              ^^^^^^^
UnboundLocalError: cannot access local variable 'classes' where it is not associated with a value
```

**Pull request merged**